11-query/04-delete-query.md

```diff
@@ -5,30 +5,35 @@ longTailKeywords: grakn delete data, graql delete query, graql delete instances
 Summary: Delete queries in Grakn.
 ---
 
-## Delete Instances of an Entity Type
-To delete an instance of an entity type from the knowledge graph, we use a [match clause](../11-query/01-match-clause.md) followed by the `delete` keyword and the variable to be deleted. To try the following examples with one of the Grakn clients, follows these [Clients Guide](#clients-guide).
+## Delete Instances of a Type
+To delete an instance of a type from the knowledge graph, we use a [match clause](../11-query/01-match-clause.md) followed by the `delete` keyword and statements indicating data to delete.
+To try the following examples with one of the Grakn clients, follows these [Clients Guide](#clients-guide).
+
+Match-Delete queries are NOT lazy: the `match` will be fully evaluated and answers recorded, and each answer will in turn have
+the operations specified in the `delete` clause applied. This avoids modifying the graph while traversing it.
 
 <div class="tabs dark">
 
 [tab:Graql]
 ```graql
-match $p isa person, has email "raphael.santos@gmail.com"; delete $p;
+match $p isa person, has email "raphael.santos@gmail.com"; delete $p isa person;
 ```
 [tab:end]
 
 [tab:Java]
 ```java
 GraqlDelete query = Graql.match(
     var("p").isa("person").has("email", "raphael.santos@gmail.com")
-).delete("p");
+).delete(var("p").isa("person"));
 ```
 [tab:end]
 </div>
 
-This deletes a particular instance of the `person` type with the id of `V41016`.
+This deletes a particular instance of the `person` type with the id of `V41016`. We must write the `isa` clause to tell
```

Comment:
I think it should be 

"This deletes a particular instance of the `person` type with the email of `raphael.santos@gmail.com`."

Reply:
thanks